filesystem/flysystem/FlysystemAssetStore.php

```diff
@@ -110,6 +110,9 @@ public function setPublicFilesystem(Filesystem $filesystem) {
 	 * @return Filesystem
 	 */
 	public function getPublicFilesystem() {
+		if(!$this->publicFilesystem) {
+			throw new Exception("Filesystem misconfiguration error");
```

Comment:
"Our own exception classes" gets messy quickly - do you have one or more per class? Or per subsystem, e.g. all `SilverStripe/Forms` namespace? I find you get most of the value out of determining an "app-level" vs. "PHP error" exception. Although for specific utilities like Guzzle, it's very handy to have library specific exceptions.